coriolis/tests/minion_manager/rpc/test_tasks.py

```diff
@@ -0,0 +1,2009 @@
+            mock.sentinel.context, mock.sentinel.action_id, {})
+
+    @mock.patch.object(
+        tasks.MinionManagerTaskEventMixin, '_get_minion_machine'
+    )
+    def test_execute_raises_exception_when_required_properties_are_missing(
+            self, mock_get_minion_machine):
+        mock_get_minion_machine.return_value = self.minion_machine
+        self.minion_machine.provider_properties = None
+
+        self.assertRaises(
+            exception.InvalidMinionMachineState,
+            self.task.execute, mock.sentinel.context)
+
+        mock_get_minion_machine.assert_called_once_with(
+            mock.sentinel.context, mock.sentinel.origin_minion_id,
+            raise_if_not_found=True)
+
+    @mock.patch.object(
+        tasks._BaseConfirmMinionAllocationForActionTask,
+        '_confirm_machine_allocation_for_action'
+    )
+    @mock.patch.object(
+        tasks._BaseConfirmMinionAllocationForActionTask, '_get_action_label'
+    )
+    @mock.patch.object(
+        tasks.MinionManagerTaskEventMixin, '_get_minion_machine'
+    )
+    def test_execute_with_exception_not_found(
+            self, mock_get_minion_machine, mock_get_action_label,
+            mock_confirm_allocation):
+        mock_get_minion_machine.return_value = self.minion_machine
+        mock_confirm_allocation.side_effect = exception.NotFound()
+
+        self.assertRaises(
+            exception.MinionMachineAllocationFailure,
+            self.task.execute, mock.sentinel.context)
+
+        mock_get_action_label.assert_called_once_with()
+
+    @mock.patch.object(
+        tasks._BaseConfirmMinionAllocationForActionTask, '_get_action_label'
+    )
+    @mock.patch.object(
+        tasks._BaseConfirmMinionAllocationForActionTask,
+        '_confirm_machine_allocation_for_action'
+    )
+    @mock.patch.object(
+        tasks.MinionManagerTaskEventMixin, '_get_minion_machine'
+    )
+    def test_execute_raises_exception_when_invalid_migration_state(
+            self, mock_get_minion_machine, mock_confirm_allocation,
+            mock_get_action_label):
+        mock_get_minion_machine.return_value = self.minion_machine
+        mock_confirm_allocation.side_effect = [
+            exception.InvalidReplicaState(reason='Invalid state')]
+
+        self.assertRaises(
+            exception.MinionMachineAllocationFailure,
+            self.task.execute, mock.sentinel.context)
+
+        mock_get_action_label.assert_called_once_with()
+
+
+class ConfirmMinionAllocationForMigrationTaskTestCase(
+        test_base.CoriolisBaseTestCase):
+
+    def setUp(self):
+        super(ConfirmMinionAllocationForMigrationTaskTestCase, self).setUp()
+        self.action_id = mock.sentinel.action_id
+        self.allocated_machine_id_mappings = [
+            mock.sentinel.allocate_machine_id_mappings]
+
+        self.task = tasks.ConfirmMinionAllocationForMigrationTask(
+            mock.sentinel.action_id, self.allocated_machine_id_mappings)
```

Comment:
You don't use `self.allocated_machine_id_mappings` anywhere. You can pass that sentinel directly
```suggestion
        self.task = tasks.ConfirmMinionAllocationForMigrationTask(
            mock.sentinel.action_id,
            mock.sentinel.allocated_machine_id_mappings)
```